### PR TITLE
Code(feat): Adding Player in-flight information

### PR DESCRIPTION
**Pull request by Zitchas**
**Feature:**

## Feature Details
Adds indicators on the right side that inform the player of their flagship's current:
- Speed
- Max acceleration (will reveal slowing when present)
- Max turning speed (will also reveal slowing when present)
- Current effective ramscoop value
- Current effective solar energy value

Flagship motion indicator
- with preference to allow choice between off, ghost, white arrow, or both.

Currently work in progress:
- [x] Longitudional thrust bar (forwards and reverse)
- [x] Rotational thrust bar (steering left/right)
- [x] Lateral thrust bar (left and right lateral thrust)
- [x] Thrust boxes and icon for thrust indicators

## UI Screenshots
Shuttle with both solar and ramscoop and both velocity indicators

### CODE REVIEW

File: source/Preferences.cpp [hunk 1]
```diff
@@ -546,6 +552,54 @@ bool Preferences::DoAlertHelper(Preferences::AlertIndicator toDo)
...
+
+
+bool Preferences::DoFlagshipVelocityIndicatorHelper(Preferences::FlagshipVelocityIndicator toDo)
+{
+	auto value = GetFlagshipVelocityIndicator();
+	if(value == FlagshipVelocityIndicator::BOTH)
+		return true;
+	else if(value == toDo)
+		return true;
+	return false;
```
Comment (RisingLeaf) [inline source/Preferences.cpp:599]:
```suggestion
	return value == FlagshipVelocityIndicator::BOTH || value == toDo;
```

File: source/Preferences.cpp [hunk 2]
```diff
@@ -546,6 +552,54 @@ bool Preferences::DoAlertHelper(Preferences::AlertIndicator toDo)
...
+}
+
+
```
Comment (RisingLeaf) [inline source/Preferences.cpp:602]:
```suggestion


```

File: source/Ship.h [hunk 3]
```diff
@@ -309,6 +309,12 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	double Hull() const;
 	double Fuel() const;
 	double Energy() const;
+	double DisplaySolar() const;
+	double DisplayRamScoop() const;
+	// These are for the thruster activity bars
+	double DisplayThrust() const;
```
Comment (RisingLeaf) [inline source/Ship.h:315]: remove the definition of the function beneath to avoid overloading error

Comment (Zitchas) [reply source/Ship.h:315]: Thanks! done.

**Review by RisingLeaf — APPROVED** [review]
good enough

**Comment by Zitchas** [comment]
Alright, everything I was hoping to get into this PR is now complete. Now we have thrust bars, whose visibility is also controlled by the "Show flagship data in HUD" preference.

Thrust Bars in action

One thing that this is important for is allowing the player and developer to see when lateral thrust is being applied. The automatic use of lateral thrust to stabilize flight (which can be suppressed by holding shift) scales in and out as needed, and is largely invisible if one doesn't have flares (which most ships do not).

Also worth noting that these bars are scales: Their full extent to the left or right indicates full strength, but half strength (via the control key) or variable strengths (via the AI & automatic systems) will be shown accordingly.

**Pull request merged**